recipes/stlab/1.7.0_and_above/conanfile.py

```diff
@@ -0,0 +1,213 @@
+            if self.settings.os != "Emscripten":
+                raise ConanInvalidConfiguration("{}/{} thread_system=pthread-emscripten is not supported on {}. Try using thread_system=auto".format(self.name, self.version, self.settings.os))
+
+    def _validate_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+        if self.options.use_boost: return
+        #
+        # On Apple we have to force the usage of boost.variant and boost.optional, because Apple's implementation of C++17
+        # is not complete.
+        #
+        raise ConanInvalidConfiguration("Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead. Try -o use_boost=True.")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, 17)
+
+        if self.info.settings.compiler == "gcc" and Version(self.info.settings.compiler.version) < "9":
+            raise ConanInvalidConfiguration("Need GCC >= 9")
+
+        if self.info.settings.compiler == "clang" and Version(self.info.settings.compiler.version) < "8":
+            raise ConanInvalidConfiguration("Need Clang >= 8")
+
+        if self.info.settings.compiler == "Visual Studio" and Version(self.info.settings.compiler.version) < "15.8":
+            raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
+
+        # Actually, we want *at least* 15.8 (MSVC 19.15), but we cannot check this for now with Conan.
+        if self.info.settings.compiler == "msvc" and Version(self.info.settings.compiler.version) < "19.15":
+            raise ConanInvalidConfiguration("Need msvc >= 19.15")
+
+        self._validate_task_system()
+        self._validate_thread_system()
+        self._validate_boost_components()
+
+    def configure(self):
+        if self.options.task_system == "auto":
+            self.options.task_system = self._default_task_system()
+
+        if self.options.thread_system == "auto":
+            self.options.thread_system = self._default_thread_system()
+
+        self.output.info("STLab Use Boost: {}.".format(self.options.use_boost))
+        self.output.info("STLab Future Coroutines: {}.".format(self.options.future_coroutines))
+        self.output.info("STLab No Standard Coroutines: {}.".format(self.options.no_std_coroutines))
+        self.output.info("STLab Task System: {}.".format(self.options.task_system))
+        self.output.info("STLab Thread System: {}.".format(self.options.thread_system))
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        tc.variables["BUILD_TESTING"] = self.options.test
+        tc.variables["STLAB_USE_BOOST_CPP17_SHIMS"] = self.options.use_boost
```

Comment:
What Cpp standard is used if boost is provided? 14? Can we lower the requirement above in validate?